pkg/prom/query.go

```diff
@@ -172,18 +184,24 @@ func runQuery(query string, ctx *Context, resCh QueryResultsChan, profileLabel s
 }
 
 // RawQuery is a direct query to the prometheus client and returns the body of the response
-func (ctx *Context) RawQuery(query string) ([]byte, error) {
+func (ctx *Context) RawQuery(query string, t time.Time) ([]byte, error) {
 	u := ctx.Client.URL(epQuery, nil)
 	q := u.Query()
 	q.Set("query", query)
 
-	// for non-range queries, we set the timestamp for the query to time-offset
-	// this is a special use case that's typically only used when our primary
-	// prom db has delayed insertion (thanos, cortex, etc...)
-	if promQueryOffset != 0 && ctx.name != AllocationContextName {
-		q.Set("time", time.Now().Add(-promQueryOffset).UTC().Format(time.RFC3339))
+	if !t.IsZero() {
```

Comment:
Why not make `t` a pointer or have a separate method (like you're doing with `QueryAtTime`) if its optional?

Reply:
I think its _okay_ to do this (when is anyone actually going to try to run a query for January 1, year 1?) but it feels awkward to my Go-sense :sweat_smile:

Reply:
But nil pointers are scary, too! It's just a zero value :) But I see what you mean, of course -- I'll try it out and see what feels better.

Reply:
Ended up liking `time.Time` more, actually. (You can use `time.Now()` and you don't have to worry about `nil` pointers and you don't have to worry about mutating the value.) But we'll see how it ages! It'd be a quick change to make.